python-avd/pyavd/_eos_designs/shared_utils/misc.py

```diff
@@ -233,3 +234,86 @@ def _get_ipv4_acl_field_with_substitution(field_value: str, replacements: dict[s
             return value
 
         return field_value
+
+    def get_l3_generic_interface_bgp_neighbors(
+        self: SharedUtils,
+        l3_generic_interfaces: (
+            EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3Interfaces
+            | EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannels
+        ),
+    ) -> list:
+        """
+        Fetches bgp neighbors for given L3 interface placeholder.
+
+        Fetches bgp neighbors (list of dict) for all interfaces under given interface type.
+        'l3_generic_interfaces' is expected to be set to either property - self.l3_interfaces or self.l3_port_channels.
+        """
+        neighbors = []
+        is_l3_interface = False
+        if isinstance(l3_generic_interfaces, EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3Interfaces):
+            is_l3_interface = True
+            node_type_in_schema = "l3_interfaces"
```

Comment:
Since this is only used to error messages, I think this name makes it easier to read.
```suggestion
            schema_key = "l3_interfaces"
```

Reply:
Yes, renamed this variable for clarity.